internal/controller/prometheus_manager.go

```diff
@@ -61,28 +58,21 @@ func RegisterMetrics() {
 }
 
 // UpdateGpuSliceMetrics updates GPU slice allocation metrics
-func (r *InstasliceReconciler) IncrementTotalProcessedGpuSliceMetrics(instasliceObj inferencev1alpha1.Instaslice, nodeName, gpuID string, profile string, pod *v1.Pod) error {
-	processedSlices, err := r.calculateProfileFitOnGPU(&instasliceObj, profile, gpuID, false, pod)
-	if err != nil {
-		return fmt.Errorf("failed to calculate processed GPU slices for profile %s: %w", profile, err)
-	}
-	instasliceMetrics.processedSlices.WithLabelValues(nodeName, gpuID).Add(float64(processedSlices))
-	return nil
+func (r *InstasliceReconciler) IncrementTotalProcessedGpuSliceMetrics(nodeName, gpuID string, profile string, size int32) {
+	instasliceMetrics.processedSlices.WithLabelValues(nodeName, gpuID).Add(float64(size))
 }
 
 // UpdateGpuSliceMetrics updates GPU slice allocation metrics
-func (r *InstasliceReconciler) UpdateDeployedPodTotalMetrics(nodeName, gpuID, namespace, podname, profile string, size int32) error {
+func (r *InstasliceReconciler) UpdateDeployedPodTotalMetrics(nodeName, gpuID, namespace, podname, profile string, size int32) {
 	if namespace == "" && podname == "" && profile == "" {
```

Comment:
if you are getting this null object, this could be an InstaSlice bug, can you please elaborate why this check is needed?

Reply:
It might be an instaslice bug, I cant replicate it but sometimes metrics updates with null objects

Reply:
Adding a warn logger and running repeated tests might be a good start

Reply:
I recommend we do the code changes